aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -160,165 +177,147 @@ module post_mint_reveal_nft::minting {
+    /// If we have already called this function before, calling it again
+    /// will update the source and destination collection config.
     public entry fun set_collection_config_and_create_collection(
         admin: &signer,
-        collection_name: String,
-        collection_description: String,
-        collection_maximum: u64,
-        collection_uri: String,
-        collection_mutate_config: vector<bool>,
-        token_name_base: String,
+        source_collection_name: String,
+        source_token_name_base: String,
+        source_token_uri: String,
+        destination_collection_name: String,
+        destination_collection_uri: String,
+        destination_collection_maximum: u64,
+        destination_collection_description: String,
+        destination_token_name_base: String,
+        destination_collection_mutate_config: vector<bool>,
         royalty_payee_address: address,
         token_description: String,
         token_maximum: u64,
         token_mutate_config: vector<bool>,
         royalty_points_den: u64,
         royalty_points_num: u64,
         public_mint_limit: u64,
-    ) acquires NFTMintConfig {
-        let nft_mint_config = borrow_global_mut<NFTMintConfig>(@post_mint_reveal_nft);
-        assert!(signer::address_of(admin) == nft_mint_config.admin, error::permission_denied(ENOT_AUTHORIZED));
-
-        assert!(vector::length(&collection_mutate_config) == 3 && vector::length(&token_mutate_config) == 5, error::invalid_argument(EVECTOR_LENGTH_UNMATCHED));
+    ) acquires NFTMintConfig, CollectionConfig, SourceToken {
+        assert!(signer::address_of(admin) == @post_mint_reveal_nft, error::permission_denied(ENOT_AUTHORIZED));
+        assert!(vector::length(&destination_collection_mutate_config) == 3 && vector::length(&token_mutate_config) == 5, error::invalid_argument(EVECTOR_LENGTH_UNMATCHED));
         assert!(royalty_points_den > 0 && royalty_points_num < royalty_points_den, error::invalid_argument(EINVALID_ROYALTY_NUMERATOR_DENOMINATOR));
-        assert!(!exists<CollectionConfig>(@post_mint_reveal_nft), error::permission_denied(ECOLLECTION_ALREADY_CREATED));
 
         let nft_mint_config = borrow_global_mut<NFTMintConfig>(@post_mint_reveal_nft);
         let resource_signer = create_signer_with_capability(&nft_mint_config.signer_cap);
 
         // Create the destination collection that holds the unique art NFT.
-        create_collection(&resource_signer, collection_name, collection_description, collection_uri, collection_maximum, collection_mutate_config);
+        create_collection(&resource_signer, destination_collection_name, destination_collection_description, destination_collection_uri, destination_collection_maximum, destination_collection_mutate_config);
+
+        // If CollectionConfig already exists, update it.
+        if (exists<CollectionConfig>(@post_mint_reveal_nft)) {
+            let collection_config = borrow_global_mut<CollectionConfig>(@post_mint_reveal_nft);
```

Comment:
This looks redundant. I wonder if it's more expensive gas-wise to just do a move_from, destroy the existing collection config, and do a move_to.

Reply:
I think it's bc the BigVector I used in `CollectionConfig` is not droppable, so I cannot make `CollectionConfig` droppable and therefore cannot `move_from`

Reply:
for the rest of the structs that are droppable (e.g. `SourceToken` etc.) just wanted to make them consistent, so didn't use `move_from` either

Reply:
It's not droppable but destructable. You can just do let CollectionConfig { _ } = config?

Reply:
If it's too much effort I'm okay with the redundancy.